### ISSUE TITLE
How can I resolve dask's “unknown job_id” error?

### ISSUE THREAD

**Issue by LiterallyUniqueLogin** [dask/dask-jobqueue#402]
I'm try to run dask on my local super computer (TSCC at the San Diego Supercomputer Center) for the first time. I'm using dask_jobqueue's `PBSCluster`. I'm seeing the following error (and associated debug messages) which is stopping my code from running:

```
DEBUG:started job: 21190023
DEBUG:adding worker tcp://132.249.106.18:46023
DEBUG:job id for new worker: 21190003.tscc-mgr7
DEBUG:21190003.tscc-mgr7 is a new job or restarting worker
ERROR:Unknown job_id: 21190003.tscc-mgr7 for worker tcp://132.249.106.16:41595
```

I'm guessing this is because in some parts of the code the job id is seen as `21190023` while in others it is seen as `21190003.tscc-mgr7`, but I don't know how to get those two to sync up.

Has anyone else run into this issue? Any suggestions?

(Note: I've successfully run jobs through manually written PBS scripts on my cluster, so the underlying scheduler works. When running my code I also see the jobs dask has submitted to the queue, so they're there, dask just needs to interface with them appropriately).

Reproducing code:
```python
import logging
import subprocess as sp

# Use Dask to smooth the PBS submission process
import dask
import dask.distributed
import dask_jobqueue

# Turn on debug level logging for dask
logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.DEBUG)


@dask.delayed
def subset_strs(chrom): 
    print("do something")

def main():
    cluster = dask_jobqueue.PBSCluster(
        name="STR_subset",
        walltime="48:00:00",
        log_directory=output_dir
    )
    cluster.scale(22)
    client = dask.distributed.Client(cluster)

    jobs = []
    for chrom in range(1, 23):
        jobs.append(subset_strs(chrom))

    futures = client.compute(jobs)

    with open("~/foo.txt", 'w') as results_file:
        for chrom, future in enumerate(futures):
            chrom += 1
            result = future.result()
            print(result)

if __name__ == "__main__":
    main()
```

My config:
```yaml
jobqueue:
  pbs:
    name: dask-worker
    cores: 1
    memory: '3.9 GB'
    processes: 1
    shebang: "#!/usr/bin/env bash"

    local-directory: $TMPDIR
    log_directory: $ME/dask/logs

    resource-spec: "nodes=1:ppn=1"
    queue: hotel
    walltime: 00:30:00

    dashboard_address: ':8713'

    job-extra: ['-M myemail@ucsd.edu', '-m a', '-V']
```

**Comment by zonca** [comment]
Thanks @mrocklin 

@LiterallyUniqueLogin  I've tested myself on TSCC with the latest dask and dask-jobqueue from conda and cannot reproduce the error.

I mostly ran your files, just simplified the example:

https://gist.github.com/zonca/6fd04d769a337f0575339de29488c1e2

I also pasted the logs

**Comment by zonca** [comment]
@LiterallyUniqueLogin is it possible you have 2 python environments and there is a version mismatch?

**Comment by LiterallyUniqueLogin** [comment]
Hi @zonca . Thank you for the quick responses. After your comments this morning I realized that the versions of `dask` I pulled in through my dependency on `scikit-allel` were quite old. I've been wrangling with software versioning all day trying to get the environment into a stable point with up to date versions of `dask` - that's been less than trivial unfortunately. Once I've done that, I'll rerun the simple commands and see what happens.
